Heh, that's what happens with master on my machine:
```
[ RUN      ] TestJobManager.AddJob
/nix/store/fibk8jc2iwswn1q1d7aw98pcw2l3lzjf-bash-4.4-p12/bin/bash: line 1: 31312 Segmentation fault      /tmp/nix-build-kodi-17.1.drv-0/xbmc-17.1-Krypton/$check_program
make: *** [Makefile:618: check] Error 139
builder for ‘/nix/store/wj1a5rahrn40d3s4bgi3s9ljhx0g4zx3-kodi-17.1.drv’ failed with exit code 2
error: build of ‘/nix/store/wj1a5rahrn40d3s4bgi3s9ljhx0g4zx3-kodi-17.1.drv’ failed
The invocation of "nix-build -A kodiPlugins.pvr-hts -A kodiPlain -A kodi /home/page/dev/nixpkgs" failed
```
So it built kodi but segfaults in a test. Can you test on master and see what do you get?
Also, would be nice if you can add a small comment making it clear that we're picking the forks cause otherwise dvd playback does not work.

--- a/pkgs/applications/video/kodi/default.nix
+++ b/pkgs/applications/video/kodi/default.nix
@@ -43,6 +43,15 @@ let
     url = "https://github.com/xbmc/FFmpeg/archive/3.1.6-${rel}.tar.gz";
     sha256 = "14jicb26s20nr3qmfpazszpc892yjwjn81zbsb8szy3a5xs19y81";
   };
+  # Usage of kodi fork of libdvdnav and libdvdread is necessary for functional dvd playback:
+  libdvdnav_src = fetchurl {
+    url = "https://github.com/xbmc/libdvdnav/archive/981488f.tar.gz";
+    sha256 = "312b3d15bc448d24e92f4b2e7248409525eccc4e75776026d805478e51c5ef3d";
+  };
+  libdvdread_src = fetchurl {
+    url = "https://github.com/xbmc/libdvdread/archive/17d99db.tar.gz";
+    sha256 = "e7179b2054163652596a56301c9f025515cb08c6d6310b42b897c3ad11c0199b";
+  };
 in stdenv.mkDerivation rec {
     name = "kodi-${version}";
     version = "17.1";
@@ -92,8 +101,8 @@ in stdenv.mkDerivation rec {
         --replace "/bin/bash" "${bash}/bin/bash -ex"
       cp ${ffmpeg_3_1_6} tools/depends/target/ffmpeg/ffmpeg-3.1.6-${rel}.tar.gz
       ln -s ${libdvdcss.src} tools/depends/target/libdvdcss/libdvdcss-master.tar.gz
-      ln -s ${libdvdnav.src} tools/depends/target/libdvdnav/libdvdnav-master.tar.gz
-      ln -s ${libdvdread.src} tools/depends/target/libdvdread/libdvdread-master.tar.gz
+      cp ${libdvdnav_src} tools/depends/target/libdvdnav/libdvdnav-master.tar.gz
+      cp ${libdvdread_src} tools/depends/target/libdvdread/libdvdread-master.tar.gz
     '';
 
     preConfigure = ''

